x-pack/legacy/plugins/reporting/server/types.ts

```diff
@@ -188,15 +191,14 @@ export interface LegacySetup {
  */
 
 export interface RequestFacade {
-  getBasePath: Legacy.Request['getBasePath'];
-  getSavedObjectsClient: Legacy.Request['getSavedObjectsClient'];
-  headers: Legacy.Request['headers'];
-  params: Legacy.Request['params'];
-  payload: JobParamPostPayload | GenerateExportTypePayload;
+  getBasePath: () => string;
+  getSavedObjectsClient: () => SavedObjectsClientContract;
+  headers: Record<string, string>;
+  params: KibanaRequest['params'];
+  body: JobParamPostPayload | GenerateExportTypePayload;
```

Comment:
I think that to fully leverage `KibanaRequest` typings, you gonna need to add equivalent generic typings to your `RequestFacade`.

Current implementation hides the actual `body`, `params` and `query` types.

See the diff in `get_job_params_from_request.ts` for example:

```ts
const { savedObjectType, savedObjectId } = request.params as {
    savedObjectType: string;
    savedObjectId: string;
  };
```

Where you need to forcecast to the expected params types.

Adding the typings would be something like:

```ts
RequestFacade<Params, Query, Body> {
   params: KibanaRequest<Params, Query, Body>['params'];
   getRawRequest: () => KibanaRequest<Params, Query, Body>;
   // ...
}
```

and the associated function:

```ts
export function makeRequestFacade<Params, Query, Body>(
  context: RequestHandlerContext,
  request: KibanaRequest<Params, Query, Body>,
  basePath: IBasePath['get']
): RequestFacade<Params, Query, Body> {
```

With proper route definition, you would then be able to remove the force cast:

```ts
// inside the handler
const request = makeRequestFacade(context, req, basePath); // properly inferred from inbound request
const { savedObjectType, savedObjectId } = request.params; // no need to force-cast
```

EDIT: more globally, now that the routes has been migrated (and once the `getSavedObjectClient` method will be removed from the fascade), is there any valid reasons to keep this `RequestFacade` instead of directly using `KibanaRequest`?

Reply:
Yeah, agreed: my concern for this PR is simply getting off legacy with the least amount of changes possible, so as to avoid any potential blocks that we might encounter. Your point here about the Facade and Generic types are valid, and I think we'll look to refactor these out (including the Facade altogether) in the future. Again, didn't want to hold up the PR with any major refactors.